contributors/design-proposals/federated-resource-quota.md

```diff
@@ -0,0 +1,53 @@
+#### NOTE: This method can prevent over consumption for users accessing underlying local clusters directly, if they only work against local cluster individually. This can’t prevent the over consumption if they deploy to multiple local clusters parallelly.
+### Consider this example:
+1. Having 3 clusters A, B and C
+2. Assigned user (namespace) a quota of 10 pods
+3. Federation quota controller set quota of cluster A to 10, B to 10 and C to 10
+4. User deploys a replicaset of 5 where 3 go to cluster A and 2 go to cluster B; user now has a remaining quota of 5 pods
+5. Federation quota controller set quota of cluster A, B and C to:<br>
+    A: 8 (3+5); B: 7 (2+5);  C: 5 (0+5)
+6. Cluster D joins, federation quota controller sets quota of cluster D to 5 (0+5)
+7. Cluster B leaves, federation quota controller does nothing.
+8. 2 replicas in cluster B rescheduled to cluster C (1 replica) and D (1 replica)
+9. Federation quota controller updates quota to:<br>
+    A: 8 (3+5); C: 6 (1+5); D: 6 (1+5)
+10. User creates another deployment with 10 pods to cluster A (5) and C (5)
+11. Federation admission controller captures it and see the request amount (10) is more than remaining quota (5), so refused the request.
+12. User creates another deployment with 2 pods to cluster A (1) and C (1)
+13. Federation quota controller updates quota to:<br>
+    A: 7 (4+3); C: 5 (2+3); D: 4 (1+3)
+14. User bypasses federation and deploy 1 pod to cluster D
+15. Federation quota controller catches this event and updates quotas to:<br>
+    A: 6 (4+2); C: 4 (2+2); D: 4 (2+2)
+16. User bypasses federation and deploy 1 pods to cluster C
+17. Federation quota controller catches this event and updates quotas to:<br>
+    A: 5 (4+1); C: 4 (3+1); D: 3 (2+1)
+18. User bypasses federation and deploy 1 pod to cluster C, and 1 pod to cluster D simultaneously. Since both clusters have 1 remaining quota which satisfy the request, both pods are deployed successfully.
+19. Federation quota controller catches this event and updates quotas to:<br>
+    A: 4 (4+0); C: 4 (4+0); D: 3 (3+0)
+20. The total used quota count is 11, which is 1 more than the original quota created in Federation Resource Quota object. This is the result of step 18 when user sent deployment request to different clusters parallelly. But since the remaining quota has been trimmed to 0, no more deployment can be executed either from federation level or through local cluster directly.
+
+### Checking admit policy on Federation Control Plane (FCP)
```

Comment:
You are right, and FCP is not aware about the replica counts as per this design proposal, however admission controller makes a lot of sense for the object count check at creation in fcp.